triton-vm/src/stark.rs

```diff
@@ -342,8 +342,15 @@ impl Stark {
             deep_codeword_components.concat(),
         )
         .unwrap();
-        let weighted_deep_codeword_components = &deep_codeword_components * &weights.deep;
-        let deep_codeword = weighted_deep_codeword_components.sum_axis(Axis(1));
+        let mut deep_codeword = Vec::with_capacity(short_domain.length);
+        deep_codeword_components
+            .axis_iter(Axis(0))
+            .into_par_iter()
+            .map(|chunk| {
+                chunk[0] * weights.deep[0] + chunk[1] * weights.deep[1] + chunk[2] * weights.deep[2]
```

Comment:
Why call the variable `chunk`? I don't think it spans more than 1 row. Maybe I am misunderstanding something ...

Reply:
Agreed. Maybe a better name is column? Or segment?